source/dem/particle_wall_broad_search.cc

```diff
@@ -49,19 +49,18 @@ ParticleWallBroadSearch<dim>::find_particle_wall_contact_pairs(
                ++particles_in_cell_iterator)
             {
               // Making the tuple and adding it to the
-              // particle_wall_contact_candidates vector. This vector is the
+              // particle_wall_contact_candidates. This unordered map is the
               // output of this function
 
-              std::tuple map_content =
-                std::make_tuple(particles_in_cell_iterator,
-                                boundary_cells_content.normal_vector,
-                                boundary_cells_content.point_on_face,
-                                boundary_cells_content.boundary_id,
-                                boundary_cells_content.global_face_id);
-
               particle_wall_contact_candidates[particles_in_cell_iterator
                                                  ->get_id()]
-                .insert({boundary_cells_content.global_face_id, map_content});
+                .emplace(
+                  boundary_cells_content.global_face_id,
+                  std::make_tuple(particles_in_cell_iterator,
+                                  boundary_cells_content.normal_vector,
+                                  boundary_cells_content.point_on_face,
+                                  boundary_cells_content.boundary_id,
+                                  boundary_cells_content.global_face_id));
```

Comment:
Why is the global_face_id stored twice? (inside and outside of the tuple) Also, almost all of the members of the boundary_cells_content are stored (not the cell it) + particles_in_cell_it. Why not store particles_in_cell_it and boundary_cells_content?

Reply:
After some investigation, turns out there is absolutely no reason why the global_face_id should be stored in the class. I'll work to remove it everywhere within this PR since it's not much work.
I think a next PR will be to merge the broad and the fine wall search because I am not sure we are not just doing duplicate work for nothing

Reply:
This is now removed :)